test/topology/test_topology.py

```diff
@@ -138,17 +135,22 @@ async def test_nodes_with_different_smp(manager: ManagerClient, random_tables: R
     # on both nodes, then it's guaranteed that the message will be
     # processed on the same shard as the calling code.
     # In the general case, we cannot assume that this same shard guarantee holds.
+    logger.info(f'Adding --smp=3 server')
     await manager.server_add(cmdline=['--smp', '3'])
+
+    # Remove the original 3 servers, the problem is easier to reproduce with --smp values
+    # that we pick, not the (currently) default --smp=2 coming from the suite.
```

Comment:
does it fail without the verbs registration fix?

Reply:
I applied this patch to the branch:
```diff
--- a/service/raft/raft_group0.cc
+++ b/service/raft/raft_group0.cc
@@ -137,9 +137,8 @@ raft_group0::raft_group0(seastar::abort_source& abort_source,
 }
 
 future<> raft_group0::start() {
-    return smp::invoke_on_all([shard0_this=this]() {
-        init_rpc_verbs(*shard0_this);
-    });
+    init_rpc_verbs(*this);
+    co_return;
 }
 
 void raft_group0::init_rpc_verbs(raft_group0& shard0_this) {
@@ -359,9 +358,7 @@ future<group0_info> persistent_discovery::run(
 }
 
 future<> raft_group0::abort() {
-    co_await smp::invoke_on_all([this]() {
-        return uninit_rpc_verbs(_ms.local());
-    });
+    co_await uninit_rpc_verbs(_ms.local());
     co_await _shutdown_gate.close();
 }
```

and it seems that some of these --smp servers are hanging during bootstrap on joining group 0 step - which I'm guessing is because we're unable to deliver Raft messages. 3 out of 4 test runs had the hang.

Reply:
ok